pkg/network/aci_network_test.go

```diff
@@ -254,3 +257,83 @@ func TestShouldCreateSubnet(t *testing.T) {
 	}
 
 }
+
+func TestValidateNetworkConfig(t *testing.T) {
+	azConfig := auth.Config{}
+	err := azConfig.SetAuthConfig(context.TODO())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	pn := &ProviderNetwork{}
+
+	cases := []struct {
+		description   string
+		setEnvVar     func()
```

Comment:
An important test case is what will happen if pn object has one or more property sets. for example, in the following piece of code

```go
if vnetResourceGroup := os.Getenv("ACI_VNET_RESOURCE_GROUP"); vnetResourceGroup != "" {
		log.G(ctx).Debug("ACI VNet resource group env variable ACI_VNET_RESOURCE_GROUP is set")

		pn.VnetResourceGroup = vnetResourceGroup
	} else if pn.VnetResourceGroup == "" {
		return errors.New("vnet resourceGroup can not be empty please set ACI_VNET_RESOURCE_GROUP")
	}
```
We need to check the following cases:
	- `ACI_VNET_RESOURCE_GROUP ` env var is set but`VnetResourceGroup` is not set.
	- `ACI_VNET_RESOURCE_GROUP ` env var is NOT set but`VnetResourceGroup` is set.
	-  Neither of them is set.
	- Both are set.
	
We may need to add the pn obj in the test case struct and test the different behaviors.